src/WebApi.OutputCache.V2/CacheOutputAttribute.cs

```diff
@@ -242,12 +255,12 @@ public override void OnActionExecuting(HttpActionContext actionContext)
                         _webApiCache.Add(baseKey, string.Empty, cacheTime.AbsoluteExpiration);
                         _webApiCache.Add(cachekey, content, cacheTime.AbsoluteExpiration, baseKey);
```

Comment:
Yeah, I do understand your point, since we already have the response at this point, we might as well use it to update the cache.

My main concern is that, in reality, `cache-control` headers are not really intended for the server, they are really intended for clients + intermediaries (such as proxies). So they instruct clients how to handle responses in terms of caching, and they instruct proxies how to deal with requests.

Building a convoluted caching logic on the *server* side based on them is not necessarily a good idea, or I am not sure everyone would be happy to have their cache getting busted at any point by and client from anywhere in the world. If you rely on caching to do performance lift for you, this is a performance black hole.

Maybe a better idea is to have this feature controlled by a flag on the caching attribute, something like `[CacheOutput(AllowCacheBypass = true)]`